src/geoFire/geoFireUtils.ts

```diff
@@ -0,0 +1,477 @@
+ */
+export function validateKey(key: string): void {
+  let error: string;
+
+  if (typeof key !== 'string') {
+    error = 'key must be a string';
+  } else if (key.length === 0) {
+    error = 'key cannot be the empty string';
+  } else if (1 + g_GEOHASH_PRECISION + key.length > 755) {
+    // Firebase can only stored child paths up to 768 characters
+    // The child path for this key is at the least: 'i/<geohash>key'
+    error = 'key is too long to be stored in Firebase';
+  } else if (/[\[\].#$\/\u0000-\u001F\u007F]/.test(key)) {
+    // Firebase does not allow node keys to contain the following characters
+    error = 'key cannot contain any of the following characters: . # $ ] [ /';
+  }
+
+  if (typeof error !== 'undefined') {
+    throw new Error('Invalid GeoFire key \'' + key + '\': ' + error);
+  }
+};
+
+/**
+ * Validates the inputted location and throws an error if it is invalid.
+ *
+ * @param location The [latitude, longitude] pair to be verified.
+ */
+export function validateLocation(location: number[]): void {
+  let error: string;
+
+  if (!Array.isArray(location)) {
+    error = 'location must be an array';
+  } else if (location.length !== 2) {
+    error = 'expected array of length 2, got length ' + location.length;
+  } else {
+    const latitude = location[0];
+    const longitude = location[1];
+
+    if (typeof latitude !== 'number' || isNaN(latitude)) {
+      error = 'latitude must be a number';
+    } else if (latitude < -90 || latitude > 90) {
+      error = 'latitude must be within the range [-90, 90]';
+    } else if (typeof longitude !== 'number' || isNaN(longitude)) {
+      error = 'longitude must be a number';
+    } else if (longitude < -180 || longitude > 180) {
+      error = 'longitude must be within the range [-180, 180]';
+    }
+  }
+
+  if (typeof error !== 'undefined') {
+    throw new Error('Invalid GeoFire location \'' + location + '\': ' + error);
+  }
+};
+
+/**
+ * Validates the inputted geohash and throws an error if it is invalid.
+ *
+ * @param geohash The geohash to be validated.
+ */
+export function validateGeohash(geohash: string): void {
+  let error;
+
+  if (typeof geohash !== 'string') {
+    error = 'geohash must be a string';
+  } else if (geohash.length === 0) {
+    error = 'geohash cannot be the empty string';
+  } else {
+    for (const letter of geohash) {
+      if (g_BASE32.indexOf(letter) === -1) {
+        error = 'geohash cannot contain \'' + letter + '\'';
+      }
+    }
+  }
+
+  if (typeof error !== 'undefined') {
+    throw new Error('Invalid GeoFire geohash \'' + geohash + '\': ' + error);
+  }
+};
+
+/**
+ * Validates the inputted query criteria and throws an error if it is invalid.
+ *
+ * @param newQueryCriteria The criteria which specifies the query's center and/or radius.
+ * @param requireCenterAndRadius The criteria which center and radius required.
+ */
+export function validateCriteria(newQueryCriteria: any, requireCenterAndRadius: boolean = false): void {
```

Comment:
We currently use the `any` type for `newQueryCriteria` here. It seems like we could probably provide an interface to help users input the proper object here.